doc/classes/GDExtension.xml

```diff
@@ -1,49 +1,64 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="GDExtension" inherits="Resource" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
 	<brief_description>
+		A shared native library for GDExtension.
 	</brief_description>
 	<description>
+		The [GDExtension] resource type represents a set of libraries written for GDExtension. These libraries expand the existing functionality of the engine. They are handled by the [GDExtensionManager] singleton.
+		[b]Note:[/b] Do not worry about GDExtension unless you know what you are doing.
+		[b]Note:[/b] GDExtension itself is not a scripting language and has no relation to [GDScript] resources.
 	</description>
 	<tutorials>
+		<link title="GDExtension overview">$DOCS_URL/tutorials/scripting/gdextension/what_is_gdextension.html</link>
+		<link title="GDExtension example in C++">$DOCS_URL/tutorials/scripting/gdextension/gdextension_cpp_example.html</link>
 	</tutorials>
 	<methods>
 		<method name="close_library">
 			<return type="void" />
 			<description>
+				Closes the current library.
 			</description>
 		</method>
 		<method name="get_minimum_library_initialization_level" qualifiers="const">
 			<return type="int" enum="GDExtension.InitializationLevel" />
 			<description>
+				Returns the lowest level required for this extension to be properly initialized. Ideally, if this value is lower than the current level, this [GDExtension] should be restarted to ensure it is properly applied.
 			</description>
 		</method>
 		<method name="initialize_library">
 			<return type="void" />
 			<param index="0" name="level" type="int" enum="GDExtension.InitializationLevel" />
 			<description>
+				Initializes the library bound to this GDextension.
```

Comment:
Maybe:

```suggestion
				Initializes this GDExtension at the given initialization level.
```

This should probably also have a note saying that developers shouldn't call this under normal circumstance, it will be handled by `GDExtensionManager`.

I'm actually not entirely sure why we've exposed this at all? It's especially weird considering we haven't exposed `deinitialize_library()`, its paired method.

Reply:
> I'm actually not entirely sure why we've exposed this at all? It's especially weird considering we haven't exposed `deinitialize_library()`, its paired method.

It may be worth discussing this with the rest of the GDExtension team. There have been times before where Reduz & Co. accidentally exposed methods to the documentation & scripting (hence the name of the PR).
After a lot of time has passed, only then you can tell there's a flaw in the API when struggling to write documentation for these.

Reply:
I've added an item to the agenda of tonight's GDExtension meeting, to discuss if this method should even be bound in the first place